includes/class-wc-stripe-apple-pay-registration.php

```diff
@@ -241,34 +242,45 @@ public function process_apple_pay_verification() {
 	 * @since 4.0.6
 	 */
 	public function admin_notices() {
-		if ( ! $this->stripe_enabled ) {
+		if ( ! $this->stripe_enabled || ! $this->payment_request ) {
 			return;
 		}
 
 		if ( ! current_user_can( 'manage_woocommerce' ) ) {
 			return;
 		}
 
-		if ( $this->payment_request && ! empty( $this->apple_pay_verify_notice ) ) {
-			$allowed_html = array(
-				'a' => array(
-					'href'  => array(),
-					'title' => array(),
-				),
-			);
-
-			echo '<div class="error stripe-apple-pay-message"><p>' . wp_kses( make_clickable( $this->apple_pay_verify_notice ), $allowed_html ) . '</p></div>';
+		$empty_notice = empty( $this->apple_pay_verify_notice );
+		if ( $empty_notice && ( $this->apple_pay_domain_set || empty( $this->secret_key ) ) ) {
+			return;
 		}
 
 		/**
 		 * Apple pay is enabled by default and domain verification initializes
 		 * when setting screen is displayed. So if domain verification is not set,
 		 * something went wrong so lets notify user.
 		 */
-		if ( ! empty( $this->secret_key ) && $this->payment_request && ! $this->apple_pay_domain_set ) {
+		$allowed_html = array(
+			'a' => array(
+				'href'  => array(),
+				'title' => array(),
+			),
+		);
+
+		echo '<div class="error stripe-apple-pay-message">';
+
+		echo '<p>Apple Pay domain verification failed';
+		$reason = $empty_notice ? '' : ' with the following error:</p><p>' . wp_kses( make_clickable( $this->apple_pay_verify_notice ), $allowed_html );
```

Comment:
"with the following error" should be localized

Reply:
Thank you!

Fixed in 456e0e22e880a07305e621e3d330030e5da286dc.